frame/elections-phragmen/src/benchmarking.rs

```diff
@@ -401,9 +404,14 @@ benchmarks! {
 
 		let _ = fill_seats_up_to::<T>(m)?;
 		let removing = as_lookup::<T>(<Elections<T>>::members_ids()[0].clone());
+		let call = Call::<T>::remove_member { who: removing, has_replacement: false }.encode();
 	}: {
 		assert_eq!(
-			<Elections<T>>::remove_member(RawOrigin::Root.into(), removing, false).unwrap_err().error,
+			<Call<T> as Decode>::decode(&mut &*call)
```

Comment:
my interest is to avoid these manual stuff. Please use the default `_(...)` syntax here. Instead of checking the error type, check that `removing` is still in the set, or check the system events for a failed transaction.

Reply:
It seems as if a failing call using `_(...)` syntax bails out and does not run the verify function, so the reset-members-code there is not run and subsequent tests fail bc. of an invalid starting condition.

Reply:
How about adding a RAII object to the benchmarks by implementing Drop so that it does the cleanup after the benchmark?

Reply:
if it is not too much of a hassle, I'd appreciate that. If it becomes too deep of a rabbit hole I am fine without it as well.

Reply:
The rabbit hole became too deep to justify the purpose. I reverted to manual syntax.